pkg/windows/calico.go

```diff
@@ -52,147 +53,226 @@ const (
 )
 
 // Setup creates the basic configuration required by the CNI.
-func (c *Calico) Setup(ctx context.Context, dataDir string, nodeConfig *daemonconfig.Node, restConfig *rest.Config) (*CNIConfig, error) {
-	calicoKubeConfig := CalicoKubeConfig{
-		Server:               "https://127.0.0.1:6443",
-		CertificateAuthority: filepath.Join("c:\\", dataDir, "agent", "server-ca.crt"),
-		Token:                "",
-		Path:                 filepath.Join("c:\\", dataDir, "agent", CalicoKubeConfigName),
+func (c *Calico) Setup(ctx context.Context, nodeConfig *daemonconfig.Node, restConfig *rest.Config) error {
+	if err := c.initializeConfig(ctx, nodeConfig, restConfig); err != nil {
+		return err
 	}
 
-	client, err := coreClient(restConfig)
-	if err != nil {
-		return nil, err
+	if err := c.overrideCalicoConfigByHelm(restConfig); err != nil {
+		return err
 	}
-	serviceAccounts := client.CoreV1().ServiceAccounts(CalicoSystemNamespace)
 
-	req := authv1.TokenRequest{
-		Spec: authv1.TokenRequestSpec{
-			Audiences:         []string{version.Program},
-			ExpirationSeconds: pointer.Int64(60 * 60 * 24 * 365),
-		},
+	if err := c.writeConfigFiles(nodeConfig.AgentConfig.CNIConfDir, nodeConfig.AgentConfig.NodeName); err != nil {
+		return err
 	}
 
-	token, err := serviceAccounts.CreateToken(ctx, calicoNode, &req, metav1.CreateOptions{})
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to create token for service account (%s/%s)", CalicoSystemNamespace, calicoNode)
+	logrus.Info("Generating HNS networks, please wait")
+	if err := c.generateCalicoNetworks(); err != nil {
+		return err
 	}
-	calicoKubeConfig.Token = token.Status.Token
 
-	cfg := &CNIConfig{NodeConfig: nodeConfig, NetworkName: "Calico", BindAddress: nodeConfig.AgentConfig.NodeIP}
+	return nil
+}
 
-	hc, err := helm.NewFactoryFromConfig(restConfig)
+func (c *Calico) GetConfig() *CNIConfig {
```

Comment:
It doesn't appear as though this getter is necessary since the field being returned is exported.

Reply:
This is the reasoning: pebinary is using the CNI interface which defines functions "Start" and "Setup". Calico struct complies with the CNI interface and it includes the CalicoCNIConfig. However, as golang interfaces only define functions, pebinary has no way to access CalicoCNIConfig (I think). That's why I created a third function (GetConfig() ) in the CNI interface to retrieve the config.

Reply:
Does the `cni` field have to be of type `win.CNI`, or could we just make it of type `Calico`? As long as `Calico` implements the `win.CNI` interface we can refer to it as the actual type within this module.

Reply:
@brandond you mean adding calico as value for cni field when initializing pebinaryexecutor? https://github.com/rancher/rke2/blob/master/pkg/rke2/rke2_windows.go#L61-L72. I guess we could do that by using 
```
if clx.String("cni") == calico {
    cni = windows.Calico
}
```
And then:
```
	return &pebinaryexecutor.PEBinaryConfig{
		Resolver:        resolver,
		ImagesDir:       agentImagesDir,
		ManifestsDir:    agentManifestsDir,
		CISMode:         isCISMode(clx),
		CloudProvider:   cpConfig,
		DataDir:         dataDir,
		AuditPolicyFile: clx.String("audit-policy-file"),
		KubeletPath:     cfg.KubeletPath,
		DisableETCD:     clx.Bool("disable-etcd"),
		IsServer:        isServer,
		cni:                cni,
	}, nil
```
And then we need to import "github.com/rancher/rke2/windows/". Is that what you mean?

Reply:
On reflection, I guess you mean adding Calico struct directly as the value of cni in `type PEBinaryConfig struct {`. We can do that but in that case, I think we should remove: `type CNI interface {` because we don't use it anymore. TBH, as it does not seem that we will add any other CNI plugin in the foreseeable future, I think it is fine

Reply:
Let's remove it and keep it explicit for now. Generally, there's never a time to add an abstraction when there's only one implementation.

Reply:
@briandowns you're suggesting we remove the windows-specific `CNI` interface that only has the single implementation? If so I am +1 on that.

Reply:
done